### DISCUSSION TITLE
_document.tsx as functional component with getInitialProps

### DISCUSSION THREAD

**Discussion by incordios** [vercel/next.js#44514] [Help]
I'm new to Next.js and want to write _document.tsx as a functional component. The following code compiles correctly, but I'm wondering if anyone more versed in Next.js can point out anything that might break due to the way this is written.

```
import { Html, Head, Main, NextScript } from "next/document";
import Document, { DocumentContext, DocumentInitialProps } from "next/document";
import { Children } from "react";

export default function Document() {
  return (
    <Html>
      <Head />
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  )
}

MyDocument.getInitialProps = async (ctx: DocumentContext): Promise<DocumentInitialProps> => {
  const initialProps = await Document.getInitialProps(ctx);
  return {
    ...initialProps,
    styles: Children.toArray([initialProps.styles]),
  } as DocumentInitialProps;
};
```

As any FYI, I plan to add NextUI down the road:
```
import { CssBaseline } from "@nextui-org/react";
...
 <Head>{CssBaseline.flush()}</Head>
...
```

**Comment by icyJoseph** [comment]
I don't think there's anything fundamentally wrong, or specifically problematic about this approach. It'll still have the same issues `_document` has, with less options to compare with around the web. `_document` tweaking beyond examples, or HTML/BODY attributes, is very seldom necessary. I mean that you won't find similar code snippet online, at least not as much.

The main issues to consider are that `_document` won't run for pre-rendered pages, and it runs on the server only for on-demand pages, while also not being called as part of SPA transitions, IIRC. 
 
There doesn't seem anything particularly problematic about https://nextui.org/docs/guide/nextui-plus-nextjs#ssr-setup either.
